config/default/jenkins-infra.yaml

```diff
@@ -143,6 +143,24 @@ jenkins:
                           }
                         }
                       }
+                  - script: >
+                      multibranchPipelineJob('jenkins-wiki-exporter') {
+                        displayName "Wiki Exporter"
+                        branchSources {
+                          github {
+                            id('2019081603')
+                            scanCredentialsId('github-access-token')
+                            repoOwner('jenkins-infra')
+                            repository('jenkins-wiki-exporter')
+                            includes('master')
+                          }
+                        }
+                        factory {
+                          workflowBranchProjectFactory {
+                            scriptPath('Jenkinsfile')
```

Comment:
Might want this to be Jenkinsfile_k8s so that ci.jenkins.io won’t pick it up? That’s if you add any k8s specific stuff to it

Reply:
> I think the default jenkins file will still work since it should auto find a node since there are none with linux && high mem, but we will see

> Might also need to setup a dind setup like i did or something
https://github.com/halkeye-docker/docker-jenkins/blob/master/jenkins.yaml#L52

I'm hoping we don't need to make a separate Jenkins file.

Since creating this PR though, I'm wondering if we can use img to build images without dind

https://github.com/genuinetools/img#running-with-docker

Reply:
I’ve never heard of img, kaniko is usually the popular one although ive never used it

Reply:
Kaniko is a root-less docker replacement, so including running images
img is docker apis, allows to build, but not run without needing to change dockerfiles and stuff. In theory at least. I've been itching to try it, so hopefully will get to this weekend

Reply:
I'm hoping we don't need to make a separate Jenkins file.
While I like the idea very much, I don't see how we can do that as the Jenkins agent must be differently configured.

Regarding img versus kaniko they both have pros and cons and are both quite popular so I am fine to pick one.

Whatever we select we need a secret in Jenkins containing `.docker/config.json`, any better idea than `config.json < echo "$DOCKER_CONFIG" | base64 -d`

Reply:
@olblak you can use some macros to convert the json file to base64 in jcasc:
https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos/credentials#credentials-plugin

Reply:
> @olblak you can use some macros to convert the json file to base64 in jcasc
Nice it will definitely help to review secret changes